internal/s3select/select.go

```diff
@@ -298,9 +364,13 @@ func (s3Select *S3Select) getProgress() (bytesScanned, bytesProcessed int64) {
 // Open - opens S3 object by using callback for SQL selection query.
 // Currently CSV, JSON and Apache Parquet formats are supported.
 func (s3Select *S3Select) Open(getReader func(offset, length int64) (io.ReadCloser, error)) error {
+	offset, end, err := s3Select.ScanRange.StartLen()
+	if err != nil {
+		return err
+	}
 	switch s3Select.Input.format {
 	case csvFormat:
-		rc, err := getReader(0, -1)
+		rc, err := getReader(offset, end)
```

Comment:
FYI I don't believe this matches the s3 logic - from https://docs.aws.amazon.com/AmazonS3/latest/userguide/selecting-content-from-objects.html: 
`A record that starts within the scan range specified but extends beyond the scan range will be processed by the query.`

Reply:
Also just to confirm, is this going to work with search expressions? A surface read implies that the headers would no longer be available, but I haven't dug into how the reader is implemented.

Reply:
@karlanke Both assumptions are correct and will not be changed.